scipy/integrate/_quadrature.py

```diff
@@ -662,6 +664,269 @@ def simpson(y, x=None, dx=1.0, axis=-1, even='avg'):
     return result
 
 
+def _cumulative_simpson_equal_intervals(
+    y: np.ndarray, dx: np.ndarray, axis: int
+) -> np.ndarray:
+
+    y = y.swapaxes(0, axis)
+    dx = dx.swapaxes(0, axis)
+
+    # Consider a quadratic interpolation over each set of 3 adjacent points, 
+    # a, a+h1, a+h1+h2. The subinterval widths are h1 and h2 
+    # (in this case, h1 = h2 = dx)
+    
+    # Calculate integral over the h1 subintervals.
+    # Calculate for all but last subinterval of y.
+    sub_integrals_h1 = dx[:-1] / 3.0 * \
+        (5.0 * y[:-2] / 4.0 + 2.0 * y[1:-1] - y[2:] / 4.0)
+    sub_integrals_h1[1::2] = 0
+
+    # Calculate integral over the h2 subintervals.
+    # Calculate for all but first subinterval of y.
+    sub_integrals_h2 = dx[:-1] / 3.0 * \
+        (-y[:-2] / 4.0 + 2.0 * y[1:-1] + 5.0 * y[2:] / 4.0)
+    if y.shape[0] % 2 == 0:
```

Comment:
Are the code differenced from here on between the equal and unequal interval versions? If it’s the same, it may be factored out.

Reply:
The equal and unequal versions are processed the same way after the equations are computed.

I have refactored to bring out the common code into a separate function.